gr-iio/lib/fmcomms2_source_impl.cc

```diff
@@ -334,7 +334,7 @@ template <typename T>
 void fmcomms2_source_impl<T>::set_gain_mode(size_t chan, const std::string& mode)
 {
     bool is_fmcomms4 = !iio_device_find_channel(phy, "voltage1", false);
-    if ((!is_fmcomms4 && chan > 0) || chan > 1) {
+    if ((is_fmcomms4 && chan > 0) || chan > 1) {
```

Comment:
Shouldn't we be checking for the `voltage2` channel to determine if this is a single-channel device? (`[voltage0, voltage1] ^= [ch0.real,ch0.imag]` iirc)

Reply:
If we were checking the DMAs but this is w.r.t. to the PHY.

ADRV9361-Z7035 (FMComms2/3 are the same):
```
❯ iio_attr -u ip:analog.local -c ad9361-phy
dev 'ad9361-phy', channel 'altvoltage1', id 'TX_LO' (output), found 8 channel-specific attributes
dev 'ad9361-phy', channel 'voltage1' (output), found 10 channel-specific attributes
dev 'ad9361-phy', channel 'voltage0' (input), found 15 channel-specific attributes
dev 'ad9361-phy', channel 'voltage1' (input), found 15 channel-specific attributes
dev 'ad9361-phy', channel 'voltage3' (output), found 8 channel-specific attributes
dev 'ad9361-phy', channel 'altvoltage0', id 'RX_LO' (output), found 8 channel-specific attributes
dev 'ad9361-phy', channel 'voltage2' (output), found 8 channel-specific attributes
dev 'ad9361-phy', channel 'temp0' (input), found 1 channel-specific attributes
dev 'ad9361-phy', channel 'voltage0' (output), found 10 channel-specific attributes
dev 'ad9361-phy', channel 'voltage2' (input), found 13 channel-specific attributes
dev 'ad9361-phy', channel 'out' (input), found 1 channel-specific attributes
```

Pluto (FMComms4 is the same):
```
❯ iio_attr -u ip:pluto.local -c ad9361-phy
dev 'ad9361-phy', channel 'altvoltage1', id 'TX_LO' (output), found 8 channel-specific attributes
dev 'ad9361-phy', channel 'voltage0' (input), found 15 channel-specific attributes
dev 'ad9361-phy', channel 'voltage3' (output), found 8 channel-specific attributes
dev 'ad9361-phy', channel 'altvoltage0', id 'RX_LO' (output), found 8 channel-specific attributes
dev 'ad9361-phy', channel 'voltage2' (output), found 8 channel-specific attributes
dev 'ad9361-phy', channel 'temp0' (input), found 1 channel-specific attributes
dev 'ad9361-phy', channel 'voltage0' (output), found 10 channel-specific attributes
dev 'ad9361-phy', channel 'voltage2' (input), found 13 channel-specific attributes
dev 'ad9361-phy', channel 'out' (input), found 1 channel-specific attributes
```